Sources/NIO/CircularBuffer.swift

```diff
@@ -28,61 +28,69 @@ public struct CircularBuffer<Element>: CustomStringConvertible, AppendableCollec
     private var buffer: ContiguousArray<Element?>
 
     /// The index into the buffer of the first item
-    private(set) /* private but tests */ internal var headIdx = 0
+    private(set) /* private but tests */ internal var headIdx: Index
 
     /// The index into the buffer of the next free slot
-    private(set) /* private but tests */ internal var tailIdx = 0
+    private(set) /* private but tests */ internal var tailIdx: Index
 
     public struct Index: Strideable, Comparable {
-        @usableFromInline
-        var _value: Int
-
-        public typealias Stride = Int
+        @usableFromInline var _value: ContiguousArray<Element?>.Index
+        @usableFromInline var _headValue: ContiguousArray<Element?>.Index
+        @usableFromInline let underlyingCount: Int
```

Comment:
and I guess `underlyingCount` could be `bufferCount`. Or maybe we could use the word `backing`, ie. `backingIndex`, `backingIndexOfHead` and `backingCount`? I would also like `underlyingBuffer` but then all becomes a bit long

Reply:
I think we should go with `backing`, seems it's the best one.